### DISCUSSION TITLE
Equals command not working as expected

### DISCUSSION THREAD

**Discussion by joshslark** [JetBrains/ideavim#347] [Q&A]
I'm used to working with VSVim in Visual Studio and I'm migrating to JetBrains Rider with IdeaVim. One common task that I use vim emulation for is to visual select a group of lines and auto-indent them using the equals command. However, with IdeaVim in Rider, it doesn't seem to do anything. My expectation is that it would trigger the Code > Auto-Indent Lines function. Could someone advise me as to how ideavim is intended to work with the Equals command.

I'm including a link to the Vim help file to eliminate any ambiguity of what I mean by the [equals](https://vimhelp.org/change.txt.html#%3D``) command.

Thanks

**Comment by Mishkun** [comment]
https://github.com/JetBrains/ideavim/blob/master/src/com/maddyhome/idea/vim/package-info.java#L153
This documentation says that `=` command indeed formats visual selection, and is works for me in IntelliJ Idea. Did you try to trigger `Code > Auto-Indent Lines` by hand using menu to see if it does anything at all in your file?

**Reply by joshslark** [reply]
Yes, triggering it by hand achieves the desired effect, but triggering it using the `=` command is essentially a no-op. The UI acts as though I triggered a command, but nothing changes. 

I tried a couple of different experiments.  It works when I trigger it using the ex command `:action AutoIndentLines`.  I turned on action tracking and it is calling `AutoIndentLines` action. Strangely, AutoIndentLines quits working from all three triggers when action tracking is enabled.

**Reply by Mishkun** [reply]
What if you try to add `map = <Action>(AutoIndentLines)` into yout ideavimrc file?

**Reply by joshslark** [reply]
It looks like everything works properly when I add that line to my ideavimrc file. Thanks for the help!

**Reply by Mishkun** [reply]
@joshslark can you share your ideavimrc file? Maybe some odd plugin is overriding this behaviour?

**Reply by joshslark** [reply]
I actually didn’t have an ideavimrc file before you made the suggestion. Now, I have one with just that single line that you suggested. 

I think you should be able to reproduce the problem just by installing Jetbrains Rider with the Ideavim plugin. I believe it was happening to me on multiple computers with the initial configuration.
